source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -462,7 +393,27 @@ namespace AdaptiveCards::Rendering::Uwp
 
             if (isImageSvg)
             {
-                image.as<winrt::SvgImageSource>().UriSource(imageUrl);
+                // If we have an SVG, we need to try to parse for the image size before setting the image source
+                auto svgDocumentLoadOperation = winrt::XmlDocument::LoadFromUriAsync(imageUrl);
+
+                svgDocumentLoadOperation.Completed(
+                    [weakThis = this->get_weak(),
+                     weakImageSource = winrt::make_weak(image.as<winrt::SvgImageSource>()),
+                     imageUrl](auto const& operation, auto status) -> void
+                    {
+                        auto strongThis = weakThis.get();
+                        auto strongImageSource = weakImageSource.get();
+
+                        if (strongThis && strongImageSource)
+                        {
+                            if (status == winrt::AsyncStatus::Completed)
+                            {
+                                strongThis->ParseXmlForHeightAndWidth(operation.GetResults(), strongImageSource);
+                            }
+                            // Now that we've parsed the height, we can set the image source
+                            strongThis->SetSvgUriSource(strongImageSource, imageUrl);
```

Comment:
>SetSvgUriSource [](http://example.com/codeflow?start=40&length=15)

Shouldn't this only set the source on success?

Reply:
We typically have a best effort approach to the rendering, so even though we can't set the pixel height/width, I thought we should still display the image.

Without the pixel height/width set, the image still renders, but the sizing can be off if the SVG has fixed dimensions.

Reply:
The failure doesn't seem like it would come from the width/height, though.  It would be coming from `LoadFromUriAsync`.  If that fails then it probably means that the .svg is missing from disk.

Reply:
yup, I agree :)
the failure here will be coming from `LoadFromUriAsync` :)